src/scope/core/pipelines/wan2_1/vace/utils/weight_loader.py

```diff
@@ -0,0 +1,175 @@
+        )
+        logger.debug(
+            f"load_vace_weights_only: Model has vace_blocks: {hasattr(actual_model, 'vace_blocks')}"
+        )
+        logger.debug(
+            f"load_vace_weights_only: Model has vace_patch_embedding: {hasattr(actual_model, 'vace_patch_embedding')}"
+        )
+    else:
+        # Not a CausalVaceWanModel, check if PEFT-wrapped and unwrap if needed
+        is_peft_wrapped = hasattr(model, "peft_config") or hasattr(model, "base_model")
+        if is_peft_wrapped:
+            logger.debug(
+                "load_vace_weights_only: Detected PEFT-wrapped model (non-VACE), accessing base_model"
+            )
+
+            if hasattr(model, "base_model") and hasattr(model.base_model, "model"):
+                actual_model = model.base_model.model
+            elif hasattr(model, "base_model"):
+                actual_model = model.base_model
+
+            logger.debug(
+                f"load_vace_weights_only: Unwrapped to {type(actual_model).__name__}"
+            )
+            logger.debug(
+                f"load_vace_weights_only: Unwrapped model has vace_blocks: {hasattr(actual_model, 'vace_blocks')}"
+            )
+            logger.debug(
+                f"load_vace_weights_only: Unwrapped model has vace_patch_embedding: {hasattr(actual_model, 'vace_patch_embedding')}"
+            )
+        else:
+            logger.debug(
+                f"load_vace_weights_only: Model not PEFT-wrapped, type: {model_type_name}"
+            )
+            logger.debug(
+                f"load_vace_weights_only: Model has vace_blocks: {hasattr(model, 'vace_blocks')}"
+            )
+            logger.debug(
+                f"load_vace_weights_only: Model has vace_patch_embedding: {hasattr(model, 'vace_patch_embedding')}"
+            )
+
+    # Load full VACE checkpoint
+    state_dict = load_state_dict(vace_checkpoint_path)
+
+    # Filter to only VACE-specific keys
+    vace_keys = [
+        "vace_blocks.",
+        "vace_patch_embedding.",
+    ]
+
+    vace_state_dict = {}
+    for key, value in state_dict.items():
+        if any(key.startswith(prefix) for prefix in vace_keys):
+            vace_state_dict[key] = value
+
+    if not vace_state_dict:
+        raise ValueError(
+            f"load_vace_weights_only: No VACE-specific weights found in checkpoint. "
+            f"Expected keys starting with: {vace_keys}"
+        )
+
+    logger.info(
+        f"load_vace_weights_only: Found {len(vace_state_dict)} VACE-specific parameters"
+    )
+
+    # Check shapes before loading
+    if "vace_patch_embedding.weight" in vace_state_dict:
+        ckpt_shape = vace_state_dict["vace_patch_embedding.weight"].shape
+        model_shape = actual_model.vace_patch_embedding.weight.shape
+        logger.debug(
+            f"load_vace_weights_only: Checkpoint shape: {ckpt_shape}, Model shape: {model_shape}"
+        )
+        if ckpt_shape != model_shape:
```

Comment:
Should exit early (i.e. raise an error to be caught by caller) here if the shape mismatch means that we don't have compatible embedding weights.